src/main/scala/firrtl/ir/IR.scala

```diff
@@ -22,20 +22,45 @@ case object NoInfo extends Info {
   override def toString: String = ""
   def ++(that: Info): Info = that
 }
-case class FileInfo(info: StringLit) extends Info {
-  override def toString: String = " @[" + info.serialize + "]"
+
+/** Stores the string of a file info annotation in its escaped form. */
+case class FileInfo(escaped: String) extends Info {
+  override def toString: String = " @[" + escaped + "]"
   //scalastyle:off method.name
   def ++(that: Info): Info = if (that == NoInfo) this else MultiInfo(Seq(this, that))
+  def unescaped: String = FileInfo.unescape(escaped)
+}
```

Comment:
To be backwards compatible, this needs a constructor that accepts `info: StringLit` and an accessor for `info`: `def info: StringLit`.

Also we should think about `unapply`, anyone `unapplying` and using the result will have their code break from this change. We could manually implement the `unapply` and deprecate it; then we won't have an unapply but we can add it after removing the old one.

Reply:
> To be backwards compatible, this needs a constructor that accepts ``

I added a backwards compatible apply method. That should hopefully be enough. Do you really want to be backwards compatible for people who call the constructor directly (i.e. with `new`)?

I could try to add an unapply, but it would be pretty inefficient to use. Essentially you would have to call `unescape` for it.

Reply:
@jackkoenig : turns out at least `Scala 2.11` does not like having two `unapply` methods:
```
[error] /Volumes/develop/firrtl/src/main/scala/firrtl/ir/IR.scala:29:12: method unapply is defined twice
[error]   conflicting symbols both originated in file '/Volumes/develop/firrtl/src/main/scala/firrtl/ir/IR.scala'
```

So it seems like this would require us to make `FileInfo` not a case class.

> Also we should think about `unapply`, anyone `unapplying` and using the result will have their code break from this change. We could manually implement the `unapply` and deprecate it; then we won't have an unapply but we can add it after removing the old one.